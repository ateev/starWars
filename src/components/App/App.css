.App {
  text-align: center;
}

.App-title {
  display: inline-block;
}

.Log-out {
  display: inline-block;
  float: right;
  font-size: 12px;
}

.App-header {
  background-color: #222;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}
