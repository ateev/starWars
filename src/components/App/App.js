import React, { Component } from 'react';
import { Provider } from 'react-redux';
import MainContainer from '../MainContainer/MainContainer.js';
import { createNewStore } from '../../mainStore.js';
import LogOut from '../LogOut/LogOut.js';
import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.theStore = createNewStore();
  }
  render() {
    return (
      <Provider store={this.theStore}>
        <div className="App">
            <div className="App-header">
              <div className="App-title">Star Wars Paradise</div>
              <LogOut />
            </div>
            <MainContainer />
        </div>
      </Provider >
    );
  }
}

export default App;