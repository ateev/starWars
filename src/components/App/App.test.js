import React from 'react';
import { shallow } from 'enzyme';
import App from './App.js';

describe('<App />', () => {
  it('renders without crashing', () => {
    const component = shallow(
        <App />
    );
    expect(component).toMatchSnapshot();
  });
})