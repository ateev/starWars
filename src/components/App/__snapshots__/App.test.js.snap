exports[`<App /> renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }>
    <div
        className="App">
        <div
            className="App-header">
            <div
                className="App-title">
                Star Wars Paradise
            </div>
        </div>
        <Connect(MainContainer) />
    </div>
</Provider>,
  "nodes": Array [
    <Provider
      store={
            Object {
                  "dispatch": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                  Symbol(observable): [Function],
                }
      }>
      <div
            className="App">
            <div
                  className="App-header">
                  <div
                        className="App-title">
                        Star Wars Paradise
                  </div>
            </div>
            <Connect(MainContainer) />
      </div>
</Provider>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "theStore": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }>
          <div
                    className="App">
                    <div
                              className="App-header">
                              <div
                                        className="App-title">
                                        Star Wars Paradise
                              </div>
                    </div>
                    <Connect(MainContainer) />
          </div>
</Provider>,
        "_debugID": 2,
        "_renderedOutput": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }>
          <div
                    className="App">
                    <div
                              className="App-header">
                              <div
                                        className="App-title">
                                        Star Wars Paradise
                              </div>
                    </div>
                    <Connect(MainContainer) />
          </div>
</Provider>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
